### PR TITLE
Allow passing in device_id and device_fp

### PR DESCRIPTION
**Pull request by seriaati**

### CODE REVIEW

File: genshin/client/components/base.py
Comment (jokelbaf) [inline genshin/client/components/base.py]: Could you maybe do this instead?
```python
headers.update({k: v for k, v in {
    "x-rpc-device_id": self.device_id,
    "x-rpc-device_fp": self.device_fp
}.items() if v})
```

**Review by jokelbaf — APPROVED** [review]
Looks fine to me

**Comment by seriaati** [comment]
I'll wait for ashlen's approval too

File: genshin/client/components/base.py
Comment (ashleney) [inline genshin/client/components/base.py]: You're doing this in the hook, not in the request function. Please move it.

File: genshin/client/components/base.py
Comment (ashleney) [inline genshin/client/components/base.py]: Wouldn't it be smarter to maybe store a `headers` attribute?

**Review by ashleney — CHANGES_REQUESTED** [review]

File: genshin/client/components/base.py
Comment (ashleney) [reply genshin/client/components/base.py]: I kinda like the previous one more, there was no need for a `headers_` variable though.

**Review by ashleney — APPROVED** [review]

File: genshin/client/components/base.py
Comment (jokelbaf) [reply genshin/client/components/base.py]: Would be better to avoid more than one if that serves the same purpose

**Pull request merged**
